src/CachePlugin.php

```diff
@@ -160,6 +163,22 @@ private function getMaxAge(ResponseInterface $response)
             return (new \DateTime($header))->getTimestamp() - (new \DateTime())->getTimestamp();
         }
 
-        return $this->defaultTtl;
+        return $this->config['default_ttl'];
+    }
+
+    /**
+     * @param array $config
+     *
+     * @return array
+     */
+    protected function configure(array $config = [])
+    {
+        $resolver = new OptionsResolver();
+        $resolver->setDefaults($this->config);
```

Comment:
sounds good to me. but i would keep that method private, no need to
customize that. and assign directly in the constructor. changing plugin
config at runtime would feel odd to me